src/scripts/status.js

```diff
@@ -77,50 +75,42 @@ async function contractsInfo(appController) {
   });
 }
 
-async function stdlibInfo(appController) {
-  if (appController.isLib()) return;
+async function stdlibInfo(networkFile) {
+  if (networkFile.isLib()) return;
+  const packageFile = networkFile.packageFile
   log.info('Standard library:');
 
-  // REFACTOR: Part of this logic is duplicated from NetworkAppController#setStdilb, consider deduplicating it
-  const requiredStdlib = appController.packageData.stdlib;
-  const requiresStdlib = !_.isEmpty(appController.packageData.stdlib);
-  const networkStdlib = appController.networkPackage.stdlib;
-  const hasNetworkStdlib = !_.isEmpty(networkStdlib);
-  const networkStdlibMatches = hasNetworkStdlib && networkStdlib.name === requiredStdlib.name && networkStdlib.version === requiredStdlib.version;
-  const hasCustomDeploy = hasNetworkStdlib && networkStdlib.customDeploy;
-  const customDeployMatches = hasCustomDeploy && networkStdlibMatches;
-
-  if (!requiresStdlib) {
-    const deployedWarn = hasNetworkStdlib ? `(though ${networkStdlib.name}@${networkStdlib.version} is currently deployed)` : '';
+  if (!packageFile.hasStdlib()) {
+    const deployedWarn = networkFile.hasStdlib() ? `(though ${networkFile.stdlibName}@${networkFile.stdlibVersion} is currently deployed)` : '';
```

Comment:
Nice :-)

Reply:
Maybe in this kind of scenarios, it is easier to see how encapsulating certain functionality under an object, instead of using a plain JS one, is more helpful.